msteams-platform/toolkit/add-single-sign-on.md

```diff
@@ -0,0 +1,346 @@
+<details>
+<summary><b>Bot project
+</b></summary>
+
+1. Copy `auth/bot/public` folder to `bot/src`. The two folders contain HTML pages used for auth redirect, you need to modify `bot/src/index` file to add routing to these pages.
+
+2. Copy `auth/bot/sso` folder to `bot/src`. The two folders contain three files as reference for SSO implementation:
+
+    * `showUserInfo`: It implements a function to get user info with SSO token. You can follow this to create your own method that requires SSO token.
+
+    * `ssoDialog`: It creates a [ComponentDialog](/javascript/api/botbuilder-dialogs/componentdialog?view=botbuilder-ts-latest&preserve-view=true) that is used for SSO.
+
+    * `teamsSsoBot`: It creates a [TeamsActivityHandler](/javascript/api/botbuilder/teamsactivityhandler?view=botbuilder-ts-latest&preserve-view=true) with `ssoDialog` and add `showUserInfo` as a command that can be triggered.
+
+3. Follow the code sample and register your own command with `addCommand` in this file (optional).
+
+4. Execute `npm install isomorphic-fetch` under `bot/`.
+
+5. Execute `npm install copyfiles` under `bot/` and replace following line in package.json:
+  
+   ```JSON
+
+   "build": "tsc --build",
+
+    ```
+
+   with
+
+   ```JSON
+
+   "build": "tsc --build && copyfiles public/*.html lib/",
+
+   ```
+
+   The HTML pages used for auth redirect are copied while building this bot project.
+
+6. After you add the following files, you need to create a new `teamsSsoBot` instance in `bot/src/index` file. Replace the following code:
+
+   ```Bash
+  
+   // Process Teams activity with Bot Framework.
+   server.post("/api/messages", async (req, res) => {
+   await commandBot.requestHandler(req, res);
+   });  
+
+   ```
+
+   with
+
+   ```Bash
+
+   const handler = new TeamsSsoBot();
+   // Process Teams activity with Bot Framework.
+   server.post("/api/messages", async (req, res) => {
+       await commandBot.requestHandler(req, res, async (context)=> {
+           await handler.run(context);
+       });
+   });
+
+   ```
+
+7. Add routing in `bot/src/index` file as below:
+
+   ```Bash
+
+   server.get(
+       "/auth-*.html",
+       restify.plugins.serveStatic({
+           directory: path.join(__dirname, "public"),
+       })
+   );
+
+   ```
+
+8. Add the following lines to `bot/src/index` to import `teamsSsoBot` and `path`:
+
+   ```Bash
+
+   // For ts:
+   import { TeamsSsoBot } from "./sso/teamsSsoBot";
+   const path = require("path");
+
+   // For js:
+   const { TeamsSsoBot } = require("./sso/teamsSsoBot");
+   const path = require("path");
+
+   ```
+
+9. Register your command in the Teams app manifest. Open `templates/appPackage/manifest.template.json`, and add following lines under `command` in `commandLists` of your bot:
+
+   ```JSON
+
+   {
+       "title": "show",
+       "description": "Show user profile using Single Sign On feature"
+   }
+
+   ```
+</details>
+<details>
+<summary><b>Add a new command to the bot
+</b></summary>
+
```

Comment:
Can we add a note inside here? this instructions applies to `command bot` for now, if developers start with a `bot`, we can advice them to take a look at the bot-sso sample: https://github.com/OfficeDev/TeamsFx-Samples/tree/v2/bot-sso

Reply:
Update as mentioned.